adapters/yieldlab/yieldlab.go

```diff
@@ -225,22 +225,24 @@ func (a *YieldlabAdapter) MakeBids(internalRequest *openrtb2.BidRequest, externa
 			}
 		}
 
+		imp := a.findImp(bid.ID, internalRequest.Imp)
+
 		var bidType openrtb_ext.BidType
 		responseBid := &openrtb2.Bid{
 			ID:     strconv.FormatUint(bid.ID, 10),
 			Price:  float64(bid.Price) / 100,
-			ImpID:  internalRequest.Imp[i].ID,
+			ImpID:  imp.ID,
 			CrID:   a.makeCreativeID(req, bid),
 			DealID: strconv.FormatUint(bid.Pid, 10),
 			W:      int64(width),
 			H:      int64(height),
 		}
 
-		if internalRequest.Imp[i].Video != nil {
+		if imp.Video != nil {
 			bidType = openrtb_ext.BidTypeVideo
 			responseBid.NURL = a.makeAdSourceURL(internalRequest, req, bid)
 
-		} else if internalRequest.Imp[i].Banner != nil {
+		} else if imp.Banner != nil {
 			bidType = openrtb_ext.BidTypeBanner
 			responseBid.AdM = a.makeBannerAdSource(internalRequest, req, bid)
 		} else {
```

Comment:
Hi guscarreon, thanks for your nitpicking - highly appreciated (seriously!)! Re this nitpick, fair point. However the bid should always be either ```imp.Video``` or ```imp.Banner```, as the publisher needs to configure this beforehand in our system (see the next point as well). In case the publisher configures an adslot as a type that is not supported yet by our PBS adapter and uses it in PBS, it should become obvious very quickly and then be corrected.